internal/cmd/group_utils.go

```diff
@@ -0,0 +1,52 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/tursodatabase/turso-cli/internal"
+	"github.com/tursodatabase/turso-cli/internal/prompt"
+	"github.com/tursodatabase/turso-cli/internal/turso"
+)
+
+func RetryOnSleepingGroup(client *turso.Client, group string, mainSpinnerText string, action func() (any, error)) (any, error) {
+	mainSpinner := prompt.Spinner(mainSpinnerText)
+	result, actionErr := action()
+	if actionErr == nil {
+		return result, nil
+	}
+
+	errMsg := actionErr.Error()
+	if strings.Contains(errMsg, "group_sleeping") ||
```

Comment:
I fear `group_sleeping` will never be reached because we don't actually return or formalise the API error from the `turso` client.